make/test/JtregNativeJdk.gmk

```diff
@@ -81,9 +81,12 @@ ifeq ($(call isTargetOs, macosx), true)
   BUILD_JDK_JTREG_LIBRARIES_LIBS_libTestMainKeyWindow := \
       -framework Cocoa -framework JavaNativeFoundation
   BUILD_JDK_JTREG_EXECUTABLES_LIBS_exeJniInvocationTest := -ljli
+  BUILD_JDK_JTREG_LIBRARIES_CFLAGS_libTestDynamicStore := -ObjC
```

Comment:
Instead of "tricking" the build system of compiling an Obj-C file by masquerading it as a C file and passing compiler options, you should expose the test file for what it is, and add support in the build system to handle this. I can help you with that part.

Reply:
Fixed.